packages/apollo-gateway/src/__tests__/queryPlanCucumber.test.ts

```diff
@@ -0,0 +1,138 @@
+import gql from 'graphql-tag';
+import path from 'path';
+import {
+  GraphQLSchemaModule,
+  GraphQLSchemaValidationError,
+} from 'apollo-graphql';
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import { DocumentNode, GraphQLSchema, GraphQLError } from 'graphql';
+import { composeServices, buildFederatedSchema, normalizeTypeDefs } from '@apollo/federation';
+
+import { QueryPlan } from '../..';
+import { LocalGraphQLDataSource } from '../datasources/LocalGraphQLDataSource';
+import { buildQueryPlan, buildOperationContext } from '../buildQueryPlan';
+
+const buildQueryPlanFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/build-query-plan.feature'
+);
+
+
+const features = [
+  buildQueryPlanFeature
+];
+
+function buildLocalService(modules: GraphQLSchemaModule[]) {
+  const schema = buildFederatedSchema(modules);
+  return new LocalGraphQLDataSource(schema);
+}
+
+features.forEach((feature) => {
+  defineFeature(feature, (test) => {
+    feature.scenarios.forEach((scenario) => {
+      test(scenario.title, async ({ given, when, then }) => {
+        let query: DocumentNode;
+        let schema: GraphQLSchema;
+        let errors: GraphQLError[];
+        let queryPlan: QueryPlan;
+        let options: any = {};
+
+        const serviceMap = Object.fromEntries(
+          ['accounts', 'product', 'inventory', 'reviews', 'books', 'documents'].map(
+            serviceName => {
+              return [
+                serviceName,
+                buildLocalService([
+                  require(path.join(
+                    __dirname,
+                    '__fixtures__/schemas',
+                    serviceName,
+                  )),
+                ]),
+              ] as [string, LocalGraphQLDataSource];
+            },
+          ),
+        );
+
+        ({ schema, errors } = composeServices(
+          Object.entries(serviceMap).map(([serviceName, service]) => ({
+            name: serviceName,
+            typeDefs: normalizeTypeDefs(service.sdl()),
+          })),
+        ));
```

Comment:
The approach of building this into an object via `fromEntries` then to `entries` only to convert it back into an array seemed like it could alternatively utilize a pattern that just built the service definition list directly.  

In other words, either:

1.
	```suggestion
	        const services: ServiceDefinition[] = [
	          'accounts',
	          'product',
	          'inventory',
	          'reviews',
	          'books',
	          'documents'
	        ].map(name => {
	          const service = buildLocalService([
	            require(path.join(__dirname, '__fixtures__/schemas', name)),
	          ]);
	          return { 
	            name,
	            typeDefs: normalizeTypeDefs(service.sdl()),
	          };
	        });
	
	        ({ schema, errors } = composeServices(services));
	```

Or, a few lines less, but without `service` assignment inside the `map`

2.	```suggestion
        const services: ServiceDefinition[] = [
          'accounts',
          'product',
          'inventory',
          'reviews',
          'books',
          'documents'
        ].map(name => ({
            name,
            typeDefs: normalizeTypeDefs(
              buildLocalService([
                require(path.join(__dirname, '__fixtures__/schemas', name))
              ]).sdl()
            )
        }));

        ({ schema, errors } = composeServices(services));
	```

Reply:
beautiful 😍